### PR TITLE
Feature/hbd 6568

### PR DESCRIPTION
**Pull request by mprzypasniak99**

### CODE REVIEW

File: Handbid_iPad/Views/MainContainer/Auction/ItemView.swift [hunk 1]
```diff
@@ -71,14 +76,17 @@ struct ItemView: View {
...
 			}
 			.frame(height: 20)
+			.padding(.all, 0)
+			.lineLimit(1)
 
 			Text(item.name ?? "NaN")
 				.applyTextStyle(style: .titleLeading)
 
-			Text(item.currentPrice ?? -1, format: .currency(code: "USD"))
+			Text(item.currentPrice ?? -1, format: .currency(code: currencyCode))
```
Comment (BakJakub) [inline Handbid_iPad/Views/MainContainer/Auction/ItemView.swift:85]: Why -1, not 0 ?

**Review by BakJakub — CHANGES_REQUESTED** [review]
Tests do not pass and socket for auctions does not work

File: Handbid_iPad/Views/ChooseOrgAndAuction/Auction/AuctionCollectionCellView.swift [hunk 2]
```diff
@@ -6,14 +6,21 @@ struct AuctionCollectionCellView<T: PageProtocol>: View {
...
 	var colorScheme: ColorScheme? = nil
+	let dataManager = DataManager.shared
 	let auction: AuctionModel
 	var inspection = Inspection<Self>()
 
 	var body: some View {
 		let colorScheme = colorScheme ?? defaultColorScheme
 
 		Button(action: {
-			coordinator.push(MainContainerPage.mainContainer as! T)
```
[the remark this answers is not in the document]
Comment (mprzypasniak99) [reply Handbid_iPad/Views/ChooseOrgAndAuction/Auction/AuctionCollectionCellView.swift:16]: Changed

File: Handbid_iPad/Data/WebSocket/Processors/AuctionProcessor.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024 by Handbid. All rights reserved.
+
+class AuctionProcessor: WebSocketProcessor {
+	func process(data: Data) {
+		do {
+			let message = try WebSocketMessageModel<AuctionModel>.decode(data)
```
[the remark this answers is not in the document]
Comment (mprzypasniak99) [reply Handbid_iPad/Data/WebSocket/Processors/AuctionProcessor.swift]: work

File: Handbid_iPad/ViewModels/MainContainer/AuctionViewModel.swift [hunk 4]
```diff
@@ -1,22 +1,29 @@
...
-		              items: [
-		              	ItemModel(id: 1, name: "Test Item", categoryName: "Test",
-		              	          isDirectPurchaseItem: true, isTicket: false, isPuzzle: false,
-		              	          isAppeal: false, currentPrice: 20.0, itemCode: "123"),
-		              ]),
-	]
-
-	init(dataService: DataServiceWrapper) {
+	@Published var categories: [CategoryModel] = []
+	@Published var currencyCode: String = "USD"
```
[the remark this answers is not in the document]
Comment (mprzypasniak99) [reply Handbid_iPad/ViewModels/MainContainer/AuctionViewModel.swift]: Changed

File: Handbid_iPad/ViewModels/MainContainer/AuctionViewModel.swift [hunk 5]
```diff
@@ -1,22 +1,29 @@
...
 
+import Combine
 import SwiftUI
 
 class AuctionViewModel: ObservableObject, ViewModelTopBarProtocol {
+	private var repository: AuctionRepository
+	private var auctionId: Int = 0
 	@ObservedObject var dataService: DataServiceWrapper
 	@Published var title = "Auction Details"
 	@Published var auctionStatus = "Open"
```
[the remark this answers is not in the document]
Comment (mprzypasniak99) [reply Handbid_iPad/ViewModels/MainContainer/AuctionViewModel.swift]: Changed

File: Handbid_iPad/ViewModels/MainContainer/AuctionViewModel.swift [hunk 6]
```diff
@@ -51,4 +81,23 @@ class AuctionViewModel: ObservableObject, ViewModelTopBarProtocol {
...
+			else { return }
+			handleAuctionUpdate(auction: auction)
+		}
+		catch {
+			print(error)
+		}
+	}
+
+	private func handleAuctionUpdate(auction: AuctionModel) {
+		auctionId = auction.identity ?? 0
```
[the remark this answers is not in the document]
Comment (mprzypasniak99) [reply Handbid_iPad/ViewModels/MainContainer/AuctionViewModel.swift]: Changed

File: Handbid_iPad/Views/MainContainer/Auction/ItemView.swift (hunk 1 above)
Comment (mprzypasniak99) [reply Handbid_iPad/Views/MainContainer/Auction/ItemView.swift:85]: I assume there might be a situation when there are items with price equal to 0. -1 on the other hand won't happen normally and showing price as -1 will make it more visible that there's some issue in the logic/data

**Pull request merged**